accelerometer/processing.py

```diff
@@ -0,0 +1,202 @@
+import numpy as np
+import pandas as pd
+import scipy.stats as stats
+from scipy.signal import butter, sosfiltfilt, welch
+from scipy.stats import entropy, median_abs_deviation
+from scipy.ndimage import median_filter
+from tqdm.auto import tqdm
+tqdm.pandas()
+
+
+GRAVITY_CUTOFF_HZ = 0.5
+NOISE_CUTOFF_HZ = 20
+MAXVAL, MINVAL = 6, -6  # Note: GENEActiv has a range of +/-6g whereas Axivity and Actigraph +/-8 or more
+
+
+class Processing():
+    def __init__(self, npyFile, sampleRate, epochPeriod, epochFile):
+        self.npyFile = npyFile
+        self.sampleRate = sampleRate
+        self.epochPeriod = epochPeriod
+        self.epochFile = epochFile
+
```

Comment:
Somewhere below, we would ideally also identify 'stuck' values i.e. where the device continually records a value for an unrealistically long time period.

See https://github.com/activityMonitoring/biobankAccelerometerAnalysis/blob/master/java/AccStats.java#L333